scripts/feature_tests/upgrade/1cp_1w_bootDiskImage_cluster_upgrade.sh

```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+set -x
+
+# shellcheck disable=SC1091
+source ./common.sh
+
+echo '' >~/.ssh/known_hosts
+
+start_logging "${1}"
+
+set_number_of_master_node_replicas 1
+set_number_of_worker_node_replicas 1
+
+provision_controlplane_node
+wait_for_ctrlplane_provisioning_start
+
+CP_NODE=$(kubectl get bmh -n metal3 | grep control | grep -v ready | cut -f1 -d' ')
+echo "BareMetalHost ${CP_NODE} is in provisioning or provisioned state"
+CP_NODE_IP=$(sudo virsh net-dhcp-leases baremetal | grep "${CP_NODE}" | awk '{{print $5}}' | cut -f1 -d'/')
+
+wait_for_ctrlplane_provisioning_complete "${CP_NODE}" "${CP_NODE_IP}" "controlplane node"
+# apply CNI
+ssh -o PasswordAuthentication=no -o "StrictHostKeyChecking no" "${UPGRADE_USER}@${CP_NODE_IP}" -- kubectl apply -f https://docs.projectcalico.org/manifests/calico.yaml
+
+provision_worker_node
+wait_for_worker_provisioning_start
+
+WR_NODE=$(kubectl get bmh -n metal3 | grep 'provision' | grep -v "${CP_NODE}" | cut -f1 -d' ')
+WR_NODE_IP=$(sudo virsh net-dhcp-leases baremetal | grep "${WR_NODE}" | awk '{{print $5}}' | cut -f1 -d'/')
+
+wait_for_worker_provisioning_complete 2 "${WR_NODE}" "${WR_NODE_IP}" "Worker node"
+
+echo "Create a new metal3MachineTemplate with new node image for both controlplane and worker nodes"
+cp_Metal3MachineTemplate_OUTPUT_FILE="/tmp/cp_new_image.yaml"
+wr_Metal3MachineTemplate_OUTPUT_FILE="/tmp/wr_new_image.yaml"
+CLUSTER_UID=$(kubectl get clusters -n metal3 test1 -o json | jq '.metadata.uid' | cut -f2 -d\")
+IMG_CHKSUM=$(curl -s https://cloud-images.ubuntu.com/bionic/current/MD5SUMS | grep bionic-server-cloudimg-amd64.img | cut -f1 -d' ')
```

Comment:
We are pulling the image from the internet. And, also brings the checksum from the internet as well. 
The problem we had in the past was that checksum different from the local one. i.e. the local checksum is not in sync with the one on the internet. 

One work around would be to use the same (image + checksum) from the local, since our ultimate goal is using a different Metal3MachineTemplate

Reply:
If I understood properly, you are still using the image that is on the webserver ? If so, the image used by default is raw, so the checksum would not match. you have the qcow2 and its checksum, alongside the raw one and its checksum. we are deploying raw images only (not enough RAM to deploy qcow2 images) so the checksum should not be coming directly from internet, but come from the raw image checksum.

Reply:
oh, I see, you are now downloading the image directly from internet. Unfortunately that would work only if you have 8192 GB of RAM. 4096 is not enough to download a qcow2 image. Maybe what you could do instead is to create a link to the raw image, to have a second image (pointing to the same actual file) but with a different name ?

Reply:
We can just use the same image as before. only the template reference changes. 
However, if we want to change actual boot disk image upgrade (ubuntu 18 to 19), then we need to revisit this issue. )

Reply:
In that case you will need to download the new image, convert it to raw, calculate the checksum, and then you can use it

Reply:
Done. 
Used the same local image.